editor/app/src/main/java/ca/ubc/eml/soiltopargraphy/editor/ui/flag/Flag.java

```diff
@@ -13,15 +13,17 @@
     private float latitude;
     private float longitude;
     private InfoPanel infoPanel;
+    private int terrainId;
```

Comment:
We need to adjust the `@Entity` annotation at the top of the `Flag` class so it links to `Terrain`.
```java
@Entity(tableName = "flag_table",
        foreignKeys = @ForeignKey(entity = Terrain.class,
                                  parentColumns = "terrainId", 
                                  childColumns = "terrainId",
                                  onDelete = CASCADE))
public class Flag {
```
The `CASCADE` parameter here will automatically clean up the flags when the parent terrain is deleted, so we can remove the code for that too.